simulator/base_simulator.py

```diff
@@ -84,5 +82,13 @@ def takeAction(self, key):
             print(f'Invalid key pressed: {key}')
             return
         
-        print(f'Action: {action}')
-        self.moveToNextState(action)
\ No newline at end of file
+        self.nextStep(action)
+
+    def run(self):
```

Comment:
Will this run without a keyboard? I think for the general case we don't need user input.